GPU/Common/TextureCacheCommon.cpp

```diff
@@ -177,27 +177,39 @@ void TextureCacheCommon::GetSamplingParams(int &minFilt, int &magFilt, bool &sCl
 
 	if (!(magFilt & 1) && addr != 0 && g_Config.iTexFiltering != TEX_FILTER_FORCE_NEAREST) {
 		if (videos_.find(addr & 0x3FFFFFFF) != videos_.end()) {
+			// Enforce bilinear filtering on magnification.
 			magFilt |= 1;
-			minFilt |= 1;
 		}
 	}
 
 	// Filtering overrides
-	if (g_Config.iTexFiltering == TEX_FILTER_FORCE_LINEAR) {
-		// Only override to linear filtering if there's no alpha or color testing going on.
+	switch (g_Config.iTexFiltering) {
+	case TEX_FILTER_AUTO:
+		// Follow what the game wants. We just do a single heuristic change to avoid bleeding of wacky color test colors
+		// in higher resolution (used by some games for sprites, and they accidentally have linear filter on).
+		// This does duplicate the body of the above if, but it's clearer this way.
```

Comment:
No longer above, but the nearest case?  Not an important distinction.

-[Unknown]

Reply:
Woops, it went through a couple iterations.